src/engine/CheatMenu.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using Godot;
+
+/// <summary>
+///   Handles the opening, closing and operations of the cheat menus
+/// </summary>
+public abstract class CheatMenu : Popup
+{
+    /// <summary>
+    ///   Whether the cheat menu may be opened or not
+    /// </summary>
+    public static bool CanOpenMenu => Settings.Instance.CheatsEnabled;
+
+    /// <summary>
+    ///   Current visibility of the cheat menu
+    /// </summary>
+    /// <exception cref="InvalidOperationException">
+    ///   Thrown when you try to open the cheat menu and <see cref="CanOpenMenu">CanOpen</see> is false
+    /// </exception>
+    public bool IsMenuOpen
+    {
+        get => Visible;
+        set
+        {
+            // Is closed and cheats disabled
+            if (!CanOpenMenu && value)
+                throw new InvalidOperationException("Cheats must be enabled in the settings to open the cheat menu");
+
+            if (value)
+                Popup_();
+            else
+                Hide();
+        }
+    }
+
+    [RunOnKeyDown("g_cheat_menu")]
+    public bool ToggleCheatMenu()
+    {
+        // Is closed and cheats disabled
+        if (!CanOpenMenu && !IsMenuOpen)
+            return false;
+
+        IsMenuOpen = !IsMenuOpen;
+        return true;
+    }
+
+    public override void _EnterTree()
+    {
+        InputManager.RegisterReceiver(this);
+        CheatManager.OnHideCheatMenus += OnHideCheatMenus;
```

Comment:
I think this should additionally read the current cheat values. Because it would be confusing to do:
- Open cheat menu
- Set speed way up
- Close menu
- Reopen the menu
- See that everything should be normal, but you still have the speed boost

Reply:
That's not the case, when you follow the described procedure the menu will still have the assigned values

Reply:
The CheatMenu does not leave or enter the tree when it gets hidden or shown

Reply:
But the microbe editor would? Because that scene is destroyed while swimming around, and re-created once you re-enter the editor. So that would need to re-apply the values, right?

Reply:
Huh, I've tried entering the editor and went back to the game, but the cheats were still enabled, both, actually and in the gui

Reply:
That should not be possible... it's a huge bug (as it's unintended in the design) if the editor scene isn't actually destroyed and somehow gets reused...

Reply:
Seems like the game scene gets reused but the editor doesn't. Now I get what you mean

Reply:
Good, good, that I'm not going insane.
So to me it seems sensible that the "re-apply settings from CheatManager" is a common operation to all the cheat menus for future proofing.